src/controllers/channel.controller.js

```diff
@@ -0,0 +1,32 @@
+const Channel = require('../models/channel.model');
+
+const {
+    response_201,
+    response_400,
+    response_500
+} = require('../utils/responseCodes.utils')
+
+
+exports.updateChannel = async (req, res) => {
+    try {
+        const {name, description} = req.body;
+
+        if (!name || !description || !req.params.id) {
```

Comment:
1. It should be able to update only name or only description or both.
2. In the description of the issue it is mentioned that you are getting the whole channel object. So it better to take the id from that and no need to check for it validity

Reply:
I sort of thought the frontend will also send both anyway